src/mongo/client/dbclient_rs.cpp

```diff
@@ -124,7 +124,7 @@ ReadPreferenceSetting* _extractReadPref(const BSONObj& query, int queryOptions)
     }
 
     // Default read pref is primary only or secondary preferred with slaveOK
-    ReadPreference pref = queryOptions & QueryOption_SlaveOk
+    ReadPreference pref = (queryOptions & QueryOption_SlaveOk)
```

Comment:
Bitwise and has a higher precedence than the ternery operator, so I don't think this code needed to change.